browser/js/components/Comment/index.js

```diff
@@ -0,0 +1,207 @@
+      { (!contents.tags && isEditing) ? <RaisedButton style={buttonStyle} label="Add Tag" /> : ""}
+      { (!contents.attachments && isEditing) ? <RaisedButton style={buttonStyle} label="Add Attachment" /> : ""}
+      { (!contents.selection) ? (isEditing ? <RaisedButton style={buttonStyle} label="Add Annotation" /> : "") : (
+        <div>
+        <div>
+          <TextField
+          hintText="Annotate Code"
+          defaultValue = {contents.selection.toString()}
+          floatingLabelText="Code Annotation"
+          multiLine={true}
+           />
+          <br />
+        </div>
+        < FlatButton href="/grade">Go to Code</ FlatButton>
+        </div>
+      ) }
+      { (!contents.criteria && isEditing) ? <RaisedButton style={buttonStyle} label="Add Criteria" /> : contents.criteria }
+      { (!contents.markdown && isEditing) ? <RaisedButton style={buttonStyle} label="Add Markdown" /> : (
+        <TextField
+          hintText="Add a comment"
+          defaultValue = {contents.markdown}
+          floatingLabelText="Instructor Comments"
+          multiLine={true}
+           />
+      ) }
+
+      </div>)
+
+  }
+  editMode(){
+    this.props.dispatch({type: 'COMMENT_EDIT_START', payload: {key: this.props.commentIndex}})
+  }
+  editModeDone(){
+    this.props.dispatch({type: 'COMMENT_EDIT_DONE', payload: {key: null}})
+  }
+  render(){
+    const iconButtonElement = (
+      <IconButton
+        touch={true}
+        tooltip="edit comment"
+        tooltipPosition="bottom-left"
+        onClick={this.buttonOnClickHandler}
+      >
+        <ModeEdit color={grey400} />
+      </IconButton>
+    );
+
+    const rightIconMenu = (
+      <IconMenu iconButtonElement={iconButtonElement}>
+        <MenuItem>Reply</MenuItem>
+        <MenuItem>Forward</MenuItem>
+        <MenuItem>Delete</MenuItem>
+      </IconMenu>
+    );
+    return (
+      <Card key={1} >
+      {/*<CardHeader
+      title={this.props.contents.title}
+      actAsExpander={true}
+      showExpandableButton={true}
+      />*/}
+      <ListItem style={styles.skinny} primaryText={this.props.contents.title} primaryTogglesNestedList={true}  rightIconButton={!this.props.isEditing ? iconButtonElement : <FlatButton onClick={this.onClickDoneHandler}>Done</FlatButton> } nestedItems = {[
+          <CardText expandable={true} style={styles.noTopPadding}>
+          <hr style={styles.skinny} />
+          {this.renderComment()}
+          </CardText>
+      ]}>
+      </ListItem>
+      </Card>
+    )
+  }
+}
+
+const mapStateToProps = (state, props) => {
+  console.log('mapping state');
+  console.log('state',state);
+  console.log('props',props);
+  let nextProps = Object.assign( {}, props.contents);
+
+  nextProps.isEditing = state.comment.isEditing.key === props.commentIndex ? true : false;
+
+  if(state.annotation.selectionString != ""){
+    nextProps.selection = annotation.selection
+  }
+  console.log(nextProps);
```

Comment:
Obviously when you deploy you want to get as many of these out as possible -- I understand you guys are still building so at this point they are important!